Signal/src/ViewControllers/OWSConversationSettingsTableViewController.m

```diff
@@ -115,6 +120,58 @@ - (BOOL)isGroupThread
 - (void)configureWithThread:(TSThread *)thread
 {
     self.thread = thread;
+    [self updateEditButton];
+}
+
+- (void)updateEditButton
+{
+    OWSAssert(self.thread);
+
+    if ([self.thread isKindOfClass:[TSContactThread class]]) {
+        self.navigationItem.rightBarButtonItem =
+            [[UIBarButtonItem alloc] initWithTitle:NSLocalizedString(@"EDIT_TXT", nil)
+                                             style:UIBarButtonItemStylePlain
+                                            target:self
+                                            action:@selector(didTapEditButton)];
+    }
+}
+
+- (void)didTapEditButton
+{
+    if (![self.thread isKindOfClass:[TSContactThread class]]) {
+        DDLogError(@"%@ unexpected thread: %@ in %s", self.tag, self.thread, __PRETTY_FUNCTION__);
+        OWSAssert(NO);
+        return;
+    }
+
+    TSContactThread *contactThread = (TSContactThread *)self.thread;
+    [self.contactsViewHelper presentContactViewControllerForRecipientId:contactThread.contactIdentifier
+                                                     fromViewController:self
+                                                        editImmediately:YES];
```

Comment:
>  I tend to favor them only in contents where there is strong visual relationship between the before and after content, which I don't think is true here?

![img_0663](https://cloud.githubusercontent.com/assets/217057/25907626/eb950e0a-3575-11e7-89c1-6f76c867d1a1.PNG)
![img_0664](https://cloud.githubusercontent.com/assets/217057/25907627/eb96aef4-3575-11e7-9e03-d9f9316b8482.PNG)

Well, I'd argue that there is a pretty strong relationship between the editable portion of the before/after content, but I can table this until such a time as it makes more sense.